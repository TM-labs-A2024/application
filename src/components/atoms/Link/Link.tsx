import { Link } from '@chakra-ui/next-js'
import NextLink from 'next/link'
import { ReactNode } from 'react'

export default function LinkComponent({ href, children }: { href: string; children: ReactNode }) {
  return (
    <Link
      as={NextLink}
      href={href}
      color="green.400"
      _hover={{ color: 'green.700' }}
      data-testid="link-component"
    >
      {children}
    </Link>
  )
}
