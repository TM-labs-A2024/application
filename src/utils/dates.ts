export const formatDate = (date: string) => {
  const chunks = date.split('-')

  const newDate = chunks[1] + '-' + chunks[2] + '-' + chunks[0]

  return newDate
}
