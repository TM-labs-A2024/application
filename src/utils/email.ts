import emailjs from '@emailjs/browser'

export function sendEmail(data: {
  from_name: string
  to_name: string
  code: string | number
  to_email: string
}) {
  const templateParams = {
    ...data
  }

  const serviceKey = process.env.NEXT_PUBLIC_EMAIL_SERVICE_KEY ?? ''
  const templateKey = process.env.NEXT_PUBLIC_EMAIL_TEMPLATE_KEY ?? ''
  const apiKey = process.env.NEXT_PUBLIC_EMAIL_API_KEY ?? ''

  emailjs.send(serviceKey, templateKey, templateParams, apiKey).catch(() => {
    alert('Algo sali√≥ mal, por favor reintenta el registro.')
  })
}
