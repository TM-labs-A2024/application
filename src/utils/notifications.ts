import { GENERIC_ERROR } from '@constants/index'
import { isMobile } from '@utils/index'
import { Store } from 'react-notifications-component'

export const setupErrorNotification = () => {
  Store.addNotification(
    GENERIC_ERROR('Algo salió mal, por favor inténtelo de nuevo.', isMobile(window))
  )
}
