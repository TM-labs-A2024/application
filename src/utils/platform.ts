import { Capacitor } from '@capacitor/core'

export const isWeb = () => Capacitor.getPlatform() === 'web'

export const isIOS = () => Capacitor.getPlatform() === 'ios'

export const isAndroid = () => Capacitor.getPlatform() === 'android'
