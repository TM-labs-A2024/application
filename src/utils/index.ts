import { formatDate } from './dates'
import { sendEmail, generateVerificationCode, setupEmailSending } from './email'
import { setupErrorNotification } from './notifications'
import { isWeb, isIOS, isAndroid, isMobile } from './platform'

export {
  isWeb,
  isIOS,
  isAndroid,
  isMobile,
  sendEmail,
  generateVerificationCode,
  setupEmailSending,
  setupErrorNotification,
  formatDate
}
