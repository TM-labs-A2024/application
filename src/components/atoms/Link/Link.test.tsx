import { render, screen } from '@test/utils/index'

import '@testing-library/jest-dom'
import Link from './Link'

const props = {
  href: '/evolucion/3fa85f64-5717-4562-b3fc-2c963f66afa5/1/1'
}

const component = (
  <Link {...props}>
    <div>Child</div>
  </Link>
)

describe('Atoms > Link test', () => {
  test('The url matches the expected', async () => {
    render(component)

    const enlace = await screen.findByRole('link')
    expect(enlace).toHaveAttribute('href', props.href)
  })

  test('The children of the component are the expected ones', async () => {
    render(component)

    expect(await screen.findByText('Child')).toBeInTheDocument()
  })

  test('Matches the snapshot', async () => {
    render(component)

    expect(await screen.findByTestId('link-component')).toMatchSnapshot()
  })
})
