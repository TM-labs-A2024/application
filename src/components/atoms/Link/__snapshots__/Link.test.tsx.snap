// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms > Link test Matches the snapshot 1`] = `
<a
  class="chakra-link css-acipmx"
  data-testid="link-component"
  href="/evolucion/3fa85f64-5717-4562-b3fc-2c963f66afa5/1/1"
>
  <div>
    Child
  </div>
</a>
`
